### PR TITLE
Add roadmap draft to docs

### PR DESCRIPTION
**Pull request by stefanprodan**
Fix: #52

### CODE REVIEW

File: docs/roadmap/index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+- Design the git push API
+- Implement git push in source controller
+- Design the image scanning API
+- Implement an image scanning controller
+- Design the manifests patching component
+- Implement the image scan/patch/push workflow
+- Integrate the new components in the toolkit assembler
+- Create a migration guide from Flux annotations
+
+## The road to Helm Operator v2
```
Comment (hiddeco) [inline docs/roadmap/index.md:61]: Maybe add a note about the naming change from "Helm Operator" to "Helm controller"?

File: docs/roadmap/index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+- Offer a migration guide for those that are using Helm Operator with Helm v3 and Helm Repositories
+
+Non-Goals
+
+- Migrate users that are using Helm v2
+- Migrate users that are using Helm charts from Git
+
+Tasks
+
+- Review the helm release, chart and repository APIs
```
Comment (hiddeco) [inline docs/roadmap/index.md:76]:
```suggestion
- Review the Helm release, chart and repository APIs
```

File: docs/roadmap/index.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+- Design helm releases based on source API
+- Implement a helm controller for Helm v3 covering all the current release options
```
Comment (hiddeco) [inline docs/roadmap/index.md:78]:
```suggestion
- Implement a Helm controller for Helm v3 covering all the current release options
```

File: docs/roadmap/index.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+- Implement events in helm controller
```
Comment (hiddeco) [inline docs/roadmap/index.md:79]:
```suggestion
- Implement events in Helm controller
```

File: docs/roadmap/index.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+- Implement prometheus metrics in helm controller
```
Comment (hiddeco) [inline docs/roadmap/index.md:80]:
```suggestion
- Implement Prometheus metrics in Helm controller
```

File: docs/roadmap/index.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+- Create a migration guide for helm operator users
```
Comment (hiddeco) [inline docs/roadmap/index.md:81]:
```suggestion
- Create a migration guide for Helm Operator users
```

File: docs/roadmap/index.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+- Integrate the new components in the toolkit assembler
+- Create a migration guide from Flux annotations
+
+## The road to Helm Operator v2
+
+### Helm v3 feature parity
+
+Goals
+
+- Offer a migration guide for those that are using Helm Operator with Helm v3 and Helm Repositories
```
Comment (hiddeco) [inline docs/roadmap/index.md:67]:
```suggestion
- Offer a migration guide for those that are using Helm Operator with Helm v3 and Helm repositories
```

File: docs/roadmap/index.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+- Review the git source and kustomize APIs
+- Design the events API
+- Implement events in source and kustomize controllers
+- Implement prometheus metrics in source and kustomize controllers
+- Make the kustomize-controller apply/gc events on-par with Flux v1 apply events
+- Design the notifications and events filtering API
+- Implement a notification controller for Slack, MS Teams, Discord, Rocket
+- Implement the migration command in tk
+- Create a migration guide for flux.yaml kustomize users
```
Comment (hiddeco) [inline docs/roadmap/index.md:38]:
```suggestion
- Create a migration guide for `flux.yaml` kustomize users
```

File: docs/roadmap/index.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+- Migrate users that are using Flux to run custom scripts with flux.yaml
+- Automate the migration of flux.yaml kustomize users
+
+Tasks
+
+- Review the git source and kustomize APIs
+- Design the events API
+- Implement events in source and kustomize controllers
+- Implement prometheus metrics in source and kustomize controllers
```
Comment (hiddeco) [inline docs/roadmap/index.md:33]:
```suggestion
- Implement Prometheus metrics in source and kustomize controllers
```

File: docs/roadmap/index.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+Goals
+
+- Offer an in-place migration tool for those that are using Flux in read-only mode to synchronize plain manifests
+- Offer a migration guide for those that are using Flux in read-only mode to synchronize Kustomize overlays
+- Offer a dedicated component for forwarding events to external messaging platforms 
+
+Non-Goals
+
+- Migrate users that are using Flux to run custom scripts with flux.yaml
```
Comment (hiddeco) [inline docs/roadmap/index.md:25]:
```suggestion
- Migrate users that are using Flux to run custom scripts with `flux.yaml`
```

File: docs/roadmap/index.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+- Automate the migration of flux.yaml kustomize users
```
Comment (hiddeco) [inline docs/roadmap/index.md:26]:
```suggestion
- Automate the migration of `flux.yaml` kustomize users
```

File: docs/roadmap/index.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+    So a lot of this is still TBD - read this as our shopping list of
+    ideas after some brainstorming as Flux maintainers.
+
+## The road to Flux v2
+
+### Flux read-only feature parity
+
+This would be the first stepping stone: we want the GitOps Toolkit to be on-par with today's Flux in
+[read-only mode](https://github.com/fluxcd/flux/blob/master/docs/faq.md#can-i-run-flux-with-readonly-git-access)
+and FluxCloud notifications.
```
Comment (hiddeco) [inline docs/roadmap/index.md:15]: Maybe add a link to fluxcloud for context?

File: docs/roadmap/index.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+- Offer a dedicated component that can replace Flux v1 image update feature
+
+Non-Goals
+
+- Maintain backwards compatibility with Flux v1 annotations
+
+Tasks
+
+- Design the git push API
```
Comment (hiddeco) [inline docs/roadmap/index.md:52]:
```suggestion
- Design the Git push API
```

File: docs/roadmap/index.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+- Implement git push in source controller
```
Comment (hiddeco) [inline docs/roadmap/index.md:53]:
```suggestion
- Implement Git push in source controller
```

File: docs/roadmap/index.md (hunk 3 above)
Comment (hiddeco) [reply docs/roadmap/index.md:78]: Oh, missed the "helm" on line 77. `/s/helm/Helm/`

File: docs/roadmap/index.md (hunk 12 above)
Comment (stefanprodan) [reply docs/roadmap/index.md:15]: Please add the link so I can merge all your suggestions in one-go.

Comment (hiddeco) [reply docs/roadmap/index.md:15]:
```suggestion
and [FluxCloud](https://github.com/justinbarrick/fluxcloud) notifications.
```

File: docs/roadmap/index.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+Non-Goals
+
+- Migrate users that are using Helm v2
+- Migrate users that are using Helm charts from Git
+
+Tasks
+
+- Review the helm release, chart and repository APIs
+- Design helm releases based on source API
```
Comment (hiddeco) [inline docs/roadmap/index.md:77]:
```suggestion
- Design Helm releases based on source API
```

**Pull request merged**